File: qt/gen_qpixmapcache.cpp
// +build ignore

#include <QPixmap>
#include <QPixmapCache>
#define WORKAROUND_INNER_CLASS_DEFINITION_QPixmapCache__Key
#include <QString>
#include <QByteArray>
#include <cstring>
#include <qpixmapcache.h>
#include "gen_qpixmapcache.h"
int QPixmapCache_CacheLimit() {
	return QPixmapCache::cacheLimit();
}
void QPixmapCache_SetCacheLimit(int cacheLimit) {
	QPixmapCache::setCacheLimit(static_cast<int>(cacheLimit));
}
bool QPixmapCache_Find(struct miqt_string key, QPixmap* pixmap) {
	QString key_QString = QString::fromUtf8(key.data, key.len);
	return QPixmapCache::find(key_QString, pixmap);
}
bool QPixmapCache_Find2(const Key* key, QPixmap* pixmap) {
	return QPixmapCache::find(*key, pixmap);
}
bool QPixmapCache_Insert(struct miqt_string key, QPixmap* pixmap) {
	QString key_QString = QString::fromUtf8(key.data, key.len);
	return QPixmapCache::insert(key_QString, *pixmap);
}
Key QPixmapCache_InsertWithPixmap(QPixmap* pixmap) {
	return QPixmapCache::insert(*pixmap);
}
bool QPixmapCache_Replace(const Key* key, QPixmap* pixmap) {
	return QPixmapCache::replace(*key, *pixmap);
}
void QPixmapCache_Remove(struct miqt_string key) {
	QString key_QString = QString::fromUtf8(key.data, key.len);
	QPixmapCache::remove(key_QString);
}
void QPixmapCache_RemoveWithKey(const Key* key) {
	QPixmapCache::remove(*key);
}
void QPixmapCache_Clear() {
	QPixmapCache::clear();
}
void QPixmapCache_Delete(QPixmapCache* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QPixmapCache*>( self );
	} else {
		delete self;
	}
}
QPixmapCache__Key* QPixmapCache__Key_new() {
return new QPixmapCache::Key();
}
QPixmapCache__Key* QPixmapCache__Key_new2(const Key* other) {
return new QPixmapCache::Key(*other);
}
bool QPixmapCache__Key_OperatorEqual(const QPixmapCache__Key* self, const Key* key) {
	return (*self == *key);
}
bool QPixmapCache__Key_OperatorNotEqual(const QPixmapCache__Key* self, const Key* key) {
	return (*self != *key);
}
void QPixmapCache__Key_OperatorAssign(QPixmapCache__Key* self, const Key* other) {
	self->operator=(*other);
}
void QPixmapCache__Key_Swap(QPixmapCache__Key* self, Key* other) {
	self->swap(*other);
}
bool QPixmapCache__Key_IsValid(const QPixmapCache__Key* self) {
	return self->isValid();
}
void QPixmapCache__Key_Delete(QPixmapCache__Key* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<QPixmapCache::Key*>( self );
	} else {
		delete self;
	}
}

File: qt/gen_qpixmapcache.go
package qt

import (
	"unsafe"
)

type QPixmapCache struct {
	h          uintptr
	isSubclass bool
}

func QPixmapCache_CacheLimit() int {
	return (int)(QPixmapCache_CacheLimit())
}

func QPixmapCache_SetCacheLimit(cacheLimit int) {
	QPixmapCache_SetCacheLimit((int)(cacheLimit))
}

func QPixmapCache_Find(key string, pixmap *QPixmap) bool {
	key_ms := struct_miqt_string{}
	key_ms.data = CString(key)
	key_ms.len = size_t(len(key))
	defer free(unsafe.Pointer(key_ms.data))
	return (bool)(QPixmapCache_Find(key_ms, pixmap.cPointer()))
}

func QPixmapCache_Find2(key *Key, pixmap *QPixmap) bool {
	return (bool)(QPixmapCache_Find2(key, pixmap.cPointer()))
}

func QPixmapCache_Insert(key string, pixmap *QPixmap) bool {
	key_ms := struct_miqt_string{}
	key_ms.data = CString(key)
	key_ms.len = size_t(len(key))
	defer free(unsafe.Pointer(key_ms.data))
	return (bool)(QPixmapCache_Insert(key_ms, pixmap.cPointer()))
}

func QPixmapCache_InsertWithPixmap(pixmap *QPixmap) Key {
	xxxxxxxxx
}

func QPixmapCache_Replace(key *Key, pixmap *QPixmap) bool {
	return (bool)(QPixmapCache_Replace(key, pixmap.cPointer()))
}

func QPixmapCache_Remove(key string) {
	key_ms := struct_miqt_string{}
	key_ms.data = CString(key)
	key_ms.len = size_t(len(key))
	defer free(unsafe.Pointer(key_ms.data))
	QPixmapCache_Remove(key_ms)
}

func QPixmapCache_RemoveWithKey(key *Key) {
	QPixmapCache_RemoveWithKey(key)
}

func QPixmapCache_Clear() {
	QPixmapCache_Clear()
}

type QPixmapCache__Key struct {
	h          uintptr
	isSubclass bool
}

// NewQPixmapCache__Key constructs a new QPixmapCache::Key object.
func NewQPixmapCache__Key() *QPixmapCache__Key {
	g := newQPixmapCache__Key(QPixmapCache__Key_new())
	g.isSubclass = true
	return g
}

// NewQPixmapCache__Key2 constructs a new QPixmapCache::Key object.
func NewQPixmapCache__Key2(other *Key) *QPixmapCache__Key {
	g := newQPixmapCache__Key(QPixmapCache__Key_new2(other))
	g.isSubclass = true
	return g
}

func (this *QPixmapCache__Key) OperatorEqual(key *Key) bool {
	return (bool)(QPixmapCache__Key_OperatorEqual(this.h, key))
}

func (this *QPixmapCache__Key) OperatorNotEqual(key *Key) bool {
	return (bool)(QPixmapCache__Key_OperatorNotEqual(this.h, key))
}

func (this *QPixmapCache__Key) OperatorAssign(other *Key) {
	QPixmapCache__Key_OperatorAssign(this.h, other)
}

func (this *QPixmapCache__Key) Swap(other *Key) {
	QPixmapCache__Key_Swap(this.h, other)
}

func (this *QPixmapCache__Key) IsValid() bool {
	return (bool)(QPixmapCache__Key_IsValid(this.h))
}
